### PR TITLE
Nullable reference types

### PR DESCRIPTION
**Pull request by MrJul**
This PR enables nullable reference types in all XamlX projects.

Most Avalonia projects now use nullable annotations, but some core ones such as `Markup.Xaml.Loader` can't because XamlX isn't nullable aware. This PR fixes that.

Notable changes:
 - A couple classes with non-nullable members and a default constructor now have a constructor with parameters instead.
 - `DeclaringType` is now non-null and exposed on `IXamlMember` directly.
 - `GetMethod` overloads matching the `FindMethod` ones have been added and used in place of `FindMethod`, as most places in emit code expected methods to actually be found.
 - `TypeReferenceResolver.ResolveType` returns a non null type. There were some code paths accepting null, but at the end of the day `ResolveTypeCore` was called which throws if the type couldn't be resolved, which callers expected.
 - `CecilType.GetEnumUnderlyingType` and `CecilField.GetLiteralValue` now throw instead of returning null for non-enum/literal. This matches the SRE implementation.

### CODE REVIEW

File: Directory.Build.props [hunk 1]
```diff
@@ -2,6 +2,7 @@
 
     <PropertyGroup>
         <LangVersion>latest</LangVersion>
+        <Nullable>enable</Nullable>
```
Comment (maxkatz6) [inline Directory.Build.props:5]: I would recommend adding `<TreatWarningAsError>nullable</TreatWarningAsError>` as well

Comment (maxkatz6) [reply Directory.Build.props:5]: https://github.com/AvaloniaUI/Avalonia/blob/master/build/NullableEnable.props

File: src/XamlX/TypeSystem/TypeSystem.cs [hunk 2]
```diff
@@ -45,7 +45,7 @@ interface IXamlType : IEquatable<IXamlType>
 #if !XAMLX_INTERNAL
     public
 #endif
-    interface IXamlParameterInfo : IXamlMember
+    interface IXamlParameterInfo
```
Comment (maxkatz6) [inline src/XamlX/TypeSystem/TypeSystem.cs:48]: Is it intended?

File: src/XamlX/Parsers/SystemXamlMarkupExtensionParser/MeScanner.cs [hunk 3]
```diff
@@ -61,20 +62,20 @@ internal partial class MeScanner
```
Comment (maxkatz6) [inline src/XamlX/Parsers/SystemXamlMarkupExtensionParser/MeScanner.cs:62]: For context, this class was ported from System.XAML, but I just double checked - original wasn't ported to nullability. Unfortunately.
https://github.com/dotnet/wpf/blob/c30fa6e62d3b94e5c65dda316f5c04397a8cb70d/src/Microsoft.DotNet.Wpf/src/System.Xaml/System/Xaml/Parser/MeScanner.cs#L41

File: Directory.Build.props (hunk 1 above)
Comment (MrJul) [reply Directory.Build.props:5]: I've enabled nullable warnings as errors unconditionally (XamlX doesn't use any of the not-correctly-annotated APIs from netstandard2.0). The current PR has 0 warnings.

File: src/XamlX/TypeSystem/TypeSystem.cs (hunk 2 above)
Comment (MrJul) [reply src/XamlX/TypeSystem/TypeSystem.cs:48]: Yes, as parameter isn't really a type member. For example in SRE `ParameterInfo` doesn't derive from `MemberInfo` at all, and `DeclaringType` can't be implemented.

Sorry, I've forgot to add this to the notes!

**Review by maxkatz6 — APPROVED** [review]

**Pull request merged**